src/twisted/trial/_asynctest.py

```diff
@@ -403,3 +403,26 @@ def stop():
         finally:
             results = None
             running.pop()
+
+
+    def _waitUntilAllDisconnected(self, reactor, protocols):
+        """
+        Take a list of disconnecting protocols, callback a Deferred when
+        they're all done.
+
+        @param: The reactor to use.
+        @param: A list of L{ITransport} or similars.
+        """
+        d = defer.Deferred()
+        lc = None
+
+        def _check():
+            if not True in [x.transport.connected for x in protocols]:
+                d.callback(None)
+
+        lc = task.LoopingCall(_check)
+        lc.clock = reactor
+        lc.start(0)
```

Comment:
`0` strikes me as unduly harsh here.  in the case where this doesn't work immediately I don't see any need to burn 100% CPU in the meanwhile.  start has a flag, `now=True`, which will run it once immediately, and that will take care of the case where everything's already ready.